packages/notched-outline/index.tsx

```diff
@@ -91,49 +89,44 @@ export default class NotchedOutline extends React.Component<
     return classnames('mdc-notched-outline', Array.from(classList), className);
   }
 
-  get adapter() {
+  get adapter(): MDCNotchedOutlineAdapter {
     return {
-      getWidth: () => this.outlineElement_.current ? this.outlineElement_.current.offsetWidth : 0,
-      getHeight: () => this.outlineElement_.current ? this.outlineElement_.current.offsetHeight : 0,
       addClass: (className: string) =>
         this.setState({classList: this.state.classList.add(className)}),
       removeClass: (className: string) => {
         const {classList} = this.state;
         classList.delete(className);
         this.setState({classList});
       },
-      setOutlinePathAttr: (value: string) => {
-        if (this.pathElement_.current) {
-          this.pathElement_.current.setAttribute('d', value);
-        }
+      setNotchWidthProperty: (width: number) => {
+        this.setState({
+          actualWidth: width,
+        });
       },
-      getIdleOutlineStyleValue: (propertyName: string) => {
-        if (!this.idleElement_.current) return;
-        return window
-          .getComputedStyle(this.idleElement_.current)
-          .getPropertyValue(propertyName);
+      removeNotchWidthProperty: () => {
+        this.setState({
+          actualWidth: null,
+        });
       },
     };
   }
 
   render() {
     return (
-      <React.Fragment>
-        <div
-          className={this.classes}
-          key='notched-outline'
-          ref={this.outlineElement_}
-        >
-          <svg focusable='false'>
-            <path ref={this.pathElement_} className='mdc-notched-outline__path' />
-          </svg>
-        </div>
-        <div
-          ref={this.idleElement_}
-          className='mdc-notched-outline__idle'
-          key='notched-outline-idle'
-        />
-      </React.Fragment>
+      <div
+        className={this.classes}
+        key='notched-outline'
+        style={ {width: this.state.actualWidth ? `${this.state.actualWidth}px`: undefined} }
+        ref={this.outlineElement_}
+      >
+        <div className='mdc-notched-outline__leading'></div>
+        {this.props.children ?
```

Comment:
This should actually be:
```
<div className='mdc-notched-outline__notch'>
  {this.props.children}
</div>
```

We should always show the __notch element.

Reply:
I checked this piece of code from https://github.com/material-components/material-components-web/tree/master/packages/mdc-textfield

```html
<div class="mdc-text-field mdc-text-field--outlined mdc-text-field--no-label">
  <input type="text" class="mdc-text-field__input" aria-label="Label">
  <div class="mdc-notched-outline">
    <div class="mdc-notched-outline__leading"></div>
    <div class="mdc-notched-outline__trailing"></div>
  </div>
</div>
```

Reply:
Ok I see that now: https://github.com/material-components/material-components-web/tree/master/packages/mdc-textfield#outlined-1